### ISSUE TITLE
Kubernetes feature consumes high CPU and creates significant load on API server (consider optimizing and caching for high scale)

### ISSUE THREAD

**Issue by Dentrax** [falcosecurity/falco#2346]
**Describe the bug**

1. Falco consumes high CPU usage if Kubernetes support enabled.

2. Kubernetes API Server gets lots of `/api/v1/watch` requests from `falcosecurity-libs` UserAgent. I think this strongly affects server request duration; we removed the DaemonSet for `1h` to test and noticed many of `5-10s` req durations dropped under `<1s`.

Queries used to calculate metrics:
* CPU: `sum(irate(container_cpu_usage_seconds_total{pod=~"$daemonset_name-[a-z0-9]*",container!="",}[2m]))`
* MEM: `(sum(avg(container_memory_working_set_bytes{pod=~"$daemonset_name-[a-z0-9]*",image=""}) by (pod)))/1024^3`

Before: 

* `CPU: 161, MEM: 300, Restarts: 3100`

After remove `-k, --k8s-api` flag:

* `CPU: 9, MEM: 145, Restarts: 0`

```yaml
resources:
  limits:
    cpu: "1"
    memory: 2Gi
  requests:
    cpu: 50m
    memory: 100Mi
```

Config:

```
auditLog:
  enabled: true

falco:
  jsonOutput: true
  jsonIncludeOutputProperty: false
```

Mostly default rules have been used.

Kubernetes `v1.23.7` cluster with `350` nodes.

**How to reproduce it**

Enable Falco Kubernetes support.

**Expected behaviour**

I'm not so sure, but I expect Falco should _do_ some optimization and caching stuff to reduce overhead of API server. It should be resulting in lower CPU consumption and better reliability. I have some concerns about whether this feature is _really_ production ready.

**Screenshots**

<img width="1142" alt="Screenshot 2023-01-11 at 11 15 49" src="https://user-images.githubusercontent.com/16493751/211753316-6c693773-96ab-4e26-8692-1b925a457fb5.png">

<img width="288" alt="Screenshot 2023-01-11 at 11 14 06" src="https://user-images.githubusercontent.com/16493751/211752947-0620cb90-8897-4a07-a933-47561084663f.png">

<img width="835" alt="Screenshot 2023-01-11 at 11 16 43" src="https://user-images.githubusercontent.com/16493751/211753478-3cc0dee1-f86f-4acd-98e0-457a5a958458.png">

<img width="1693" alt="Screenshot 2023-01-11 at 11 17 20" src="https://user-images.githubusercontent.com/16493751/211753599-4f7713fa-e416-4188-a751-b4c718a2246c.png">

<img width="1272" alt="Screenshot 2023-01-11 at 11 28 48" src="https://user-images.githubusercontent.com/16493751/211755934-4fbe4e61-9113-4577-bc73-bbde1f1ac8d8.png">

**Environment**

- Falco version: 
`0.31.1`
- System info:
`-`
- Cloud provider or hardware configuration:
`OpenStack`
- OS:
`Ubuntu 20.04.4 LTS`
- Kernel: 
`Linux ... 5.15.0-46-generic #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
- Installation method:
`Kubernetes (Helm)`

**Additional context**

Slack Discussion: https://kubernetes.slack.com/archives/CMWH3EH32/p1673341804634399

/cc @developer-guy @eminaktas

**Comment by divaspathak** [comment]
Hello @Dentrax. I want to try it. Can you assign it to me?

**Comment by Dentrax** [comment]
> Hello @Dentrax. I want to try it. Can you assign it to me?

Sure, go ahead. But I don't have permission since I'm not a maintainer.

**Comment by benassipaul** [comment]
Hello,
Same issue here.

From 183MiB to 933MiB
From 240mcores to 1cores throttled by kubernetes (cpu request)

Only difference are two additionnal options -k and -K to get the k8s deployment name data.
We are forced to rollback due to the huge load of enabling the kube api connection.

Regards,

**Comment by therealdwright** [comment]
I am noticing the same issue in our cluster; as soon as we turn on Kubernetes support (large-ish cluster with hundreds of nodes), we see Falco CPU usage skyrocket from an average of `~30m` to being pinned at `~1000m`

Are we sure this feature is production-ready? It hamstrung our EKS cluster control plane due to request volume.

**Comment by Andreagit97** [comment]
@therealdwright thank you for reporting, in the last months we have seen several issues with the actual k8s implementation... for this reason we are currently working on a new k8s plugin that will replace the old implementation... this should be available in the next Falco release so 0.37.0 (end of January). BTW we will try to come out with something to test, before this date! We will keep you posted!

**Comment by Andreagit97** [comment]
We are working on this!

**Comment by Andreagit97** [comment]
The new `k8smeta` plugin should solve this issue, give it a try with our dedicated helm chart and Falco 0.36.2 https://github.com/falcosecurity/falco/issues/2973#issuecomment-1876880587

**Comment by Andreagit97** [comment]
This should be solved by [Falco 0.37.0](https://github.com/falcosecurity/falco/releases/tag/0.37.0)! Feel free to reopen if this is still an issue
